ui/index.d.ts

```diff
@@ -1256,8 +1256,10 @@ export function openReadOnlyJson(node: { label: string, fullId: string }, data:
  * @param node Typically (but not strictly) an `AzExtTreeItem`. `label` is used for the file name displayed in VS Code and `fullId` is used to uniquely identify this file
  * @param content The content to display
  * @param fileExtension The file extension
+ * @param appendDelimiter If provided, `content` will be appended to the document delimited by this string
+ * @param openInNewViewColumn Whether or not to open the read-only editor in a new `ViewColumn`
  */
-export function openReadOnlyContent(node: { label: string, fullId: string }, content: string, fileExtension: string): Promise<void>;
+export function openReadOnlyContent(node: { label: string, fullId: string }, content: string, fileExtension: string, appendDelimiter?: string, openInNewViewColumn?: boolean): Promise<void>;
```

Comment:
When you're authoring an api (which you're basically doing), it's always good to model after existing apis when possible. I feel like the closest case is `showTextDocument` which returns an `editor`. Perhaps we should do something similar - although the editor wouldn't need to be as complex - maybe just support a few methods like `append` and `clear`

A few reasons I suggest that:
- I'm not a huge fan of `appendDelimiter`. Feels like I would just add that to `content`
- Also not a huge fan of `openInNewViewColumn`. Feels like a worse knock-off of `TextDocumentShowOptions`
- Seems like your current implementation always appends. I feel like that could get unwieldy for Databases. I would think each "round" of "Execute All" should get it's own fresh results file